contracts/test/vanity_url.ts

```diff
@@ -0,0 +1,311 @@
+      await mockDC.connect(alice).register(dotName);
+    });
+
+    it("Should be able to set a new URL", async () => {
+      expect(await vanityURL.vanityURLs(tokenId, aliasName)).to.equal("");
+      expect(await vanityURL.vanityURLUpdatedAt(tokenId, aliasName)).to.equal(0);
+
+      // set a new URL
+      await vanityURL.connect(alice).setNewURL(dotName, aliasName, url, price, { value: urlUpdatePrice });
+
+      expect(await vanityURL.vanityURLs(tokenId, aliasName)).to.equal(url);
+      expect(await vanityURL.vanityURLPrices(tokenId, aliasName)).to.equal(price);
+      expect(await vanityURL.vanityURLUpdatedAt(tokenId, aliasName)).to.equal(await getTimestamp());
+    });
+
+    it("Should be able to set a new URL after the domain ownership was changed but not expired", async () => {
+      // transfer the ownership
+      await mockDC.connect(bob).trasnferDomain(dotName);
+
+      // set a new URL
+      const newAliasName = "newAliasName";
+      const newURL = "newURL";
+
+      await vanityURL.connect(bob).setNewURL(dotName, newAliasName, newURL, price, { value: urlUpdatePrice });
+
+      expect(await vanityURL.vanityURLs(tokenId, newAliasName)).to.equal(newURL);
+    });
+
+    it("Should revert if the caller is not the name owner", async () => {
+      await expect(vanityURL.setNewURL(dotName, aliasName, url, price, { value: urlUpdatePrice })).to.be.revertedWith("VanityURL: only DC owner");
+    });
+
+    it("Should revert if the URL already exists", async () => {
+      // set a new URL
+      await vanityURL.connect(alice).setNewURL(dotName, aliasName, url, price, { value: urlUpdatePrice });
+
+      // set the URL twice
+      await expect(vanityURL.connect(alice).setNewURL(dotName, aliasName, url, price, { value: urlUpdatePrice })).to.be.revertedWith("VanityURL: url already exists");
+    });
+
+    it("Should revert if the payment is insufficient", async () => {
+      await expect(vanityURL.connect(alice).setNewURL(dotName, aliasName, url, price, { value: urlUpdatePrice.sub(1) })).to.be.revertedWith("VanityURL: insufficient payment");
+    });
+
+    it("Should revert if the domain is expired", async () => {
+      // increase time
+      const duration = await mockDC.duration();
+      await increaseTime(Number(duration.add(1)));
+
+      // set a new URL
+      const newAliasName = "newAliasName";
+      const newURL = "newURL";
+
+      await expect(vanityURL.connect(alice).setNewURL(dotName, newAliasName, newURL, price, { value: urlUpdatePrice })).to.be.revertedWith("VanityURL: expired domain")
+    });
+  });
+
+  describe("deleteURL", () => {
+    const tokenId = ethers.utils.keccak256(ethers.utils.toUtf8Bytes(dotName));
+    const aliasName = "aliasName";
+    const url = "url";
+    const price = ethers.utils.parseEther("2");
+
+    beforeEach(async () => {
+      await mockDC.connect(alice).register(dotName);
+      await vanityURL.connect(alice).setNewURL(dotName, aliasName, url, price, { value: urlUpdatePrice });
```

Comment:
won't this fail if the test fails to delete the url (such as those expect.revert tests)?

Reply:
Would you elaborate on it?